File: src/main/java/io/github/tanguygab/conditionalactions/commands/CACommand.java
package io.github.tanguygab.conditionalactions.commands;

import io.github.tanguygab.conditionalactions.ConditionalActions;
import io.github.tanguygab.conditionalactions.Utils;
import lombok.AllArgsConstructor;
import org.bukkit.command.CommandSender;

import java.util.Collection;
import java.util.List;

@AllArgsConstructor
public abstract class CACommand {

    protected final ConditionalActions plugin;

    public abstract void onCommand(CommandSender sender, String[] args);
    public abstract List<String> onTabComplete(CommandSender sender, String[] args);

    protected void sendMessage(CommandSender sender, String message) {
        sender.sendMessage(Utils.colors(message));
    }

    public List<String> getArguments(Collection<String> suggestions, String arg) {
        return suggestions.stream().filter(suggestion -> suggestion.startsWith(arg.toLowerCase())).toList();
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/commands/CACommand.kt
package io.github.tanguygab.conditionalactions.commands

import io.github.tanguygab.conditionalactions.ConditionalActions
import io.github.tanguygab.conditionalactions.Utils
import net.kyori.adventure.text.minimessage.MiniMessage
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer
import org.bukkit.command.CommandSender

abstract class CACommand(protected val plugin: ConditionalActions) {

    abstract fun onCommand(sender: CommandSender, args: List<String>)
    abstract fun onTabComplete(sender: CommandSender, args: List<String>): List<String>?

    protected fun sendMessage(sender: CommandSender, message: String) {
        sender.sendMessage(LegacyComponentSerializer.legacyAmpersand().deserialize(message))
    }

    fun getArguments(suggestions: Collection<String>, arg: String) = suggestions.filter { it.startsWith(arg.lowercase()) }
}
